src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -95147,5 +95147,59 @@ public override void M<U>(B x, U y)
             // Per https://github.com/dotnet/roslyn/issues/35075 errors should be expected
             c.VerifyDiagnostics();
         }
+
+        [Fact, WorkItem(34827, "https://github.com/dotnet/roslyn/issues/34827")]
+        public void Verify34827()
+        {
+            string source = @"
+using System.Collections.Generic;
+class C
+{
+    void M()
+    {
+        IEnumerable<object> x1 = new[] { ""string"", null }; // 1
+        IEnumerable<object?> x2 = new[] { ""string"", null };
+        IList<string> x3 = new[] { ""string"", null }; // 2
+        ICollection<string?> x4 = new[] { ""string"", null };
+        IReadOnlyList<string?> x5 = new[] { ""string"" };
+    }
+}";
+            var comp = CreateNullableCompilation(source);
+            comp.VerifyDiagnostics(
+                // (7,34): warning CS8619: Nullability of reference types in value of type 'string?[]' doesn't match target type 'IEnumerable<object>'.
+                //         IEnumerable<object> x1 = new[] { "string", null }; // 1
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, @"new[] { ""string"", null }").WithArguments("string?[]", "System.Collections.Generic.IEnumerable<object>").WithLocation(7, 34),
+                // (9,28): warning CS8619: Nullability of reference types in value of type 'string?[]' doesn't match target type 'IList<string>'.
+                //         IList<string> x3 = new[] { "string", null }; // 2
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, @"new[] { ""string"", null }").WithArguments("string?[]", "System.Collections.Generic.IList<string>").WithLocation(9, 28)
+                );
+        }
+
+        [Fact, WorkItem(34827, "https://github.com/dotnet/roslyn/issues/34827")]
+        public void Verify34827_Nested()
+        {
+            string source = @"
+using System.Collections.Generic;
+class C
+{
+    void M()
+    {
+        IEnumerable<object[]> x1 = new[] { new[] { ""string"" }, null }; // 1
```

Comment:
>new[] { new[] { ""string"" }, null } [](start = 35, length = 36)

It doesn't look like this scenario is targeting nesting case, or I misunderstood what "Nested" stands for in the name of the test.

Reply:
Fixed. Thanks

---
In reply to: [281739109](https://github.com/dotnet/roslyn/pull/35541#discussion_r281739109) [](ancestors = 281739109)